tests/python/relay/test_tensorrt.py

```diff
@@ -849,10 +849,128 @@ def load_graph():
+            result_dict[(i, use_trt)] = relay_exec.evaluate()(x=x_data[:batch_size, ...], **params)
+
+    for i in range(len(batches_to_test)):
+        assert_result_matches(result_dict[(i, True)], result_dict[(i, False)])
+
+
+def test_tensorrt_dynamic_batch_conv():
+    if should_skip():
+        return
+    batches_to_test = [1, 1, 2, 3, 1, 3, 2]
+    x_shape = (relay.Any(), 32, 8, 8)
+    x_data = np.ones([max(batches_to_test)] + list(x_shape)[1:]).astype("float32")
+    k_shape = (16, 32, 3, 3)
+    params = {"kernel": np.random.uniform(-1, 1, k_shape).astype("float32")}
+    result_dict = {}
+    for use_trt in [True, False]:
+        x = relay.var("x", shape=x_shape, dtype="float32")
+        kernel = relay.var("kernel", shape=k_shape, dtype="float32")
+        out = relay.nn.conv2d(x, kernel, channels=16, kernel_size=(3, 3), groups=1)
+        f = relay.Function([x, kernel], out)
+        mod = tvm.IRModule()
+        mod["main"] = f
+        if use_trt:
+            mod = relay.tensorrt.EnableTrt(mod, params)
+        with relay.build_config(opt_level=3):
+            relay_exec = relay.create_executor("vm", mod=mod, ctx=tvm.cpu(0), target="llvm")
+
+        for i, batch_size in enumerate(batches_to_test):
+            result_dict[(i, use_trt)] = relay_exec.evaluate()(x=x_data[:batch_size, ...], **params)
+
+    for i in range(len(batches_to_test)):
+        assert_result_matches(result_dict[(i, True)], result_dict[(i, False)])
+
+
+def test_tensorrt_dynamic_dense():
+    if should_skip():
+        return
+    batches_to_test = [1, 1, 2, 3, 1, 3, 2]
+    x_shape = (relay.Any(), 25)
+    # x_shape = (2, 25)
+
+    x_data = np.ones([max(batches_to_test)] + list(x_shape)[1:]).astype("float32")
+    weight_shape = (5, 25)
+    params = {"weights": np.random.uniform(-1, 1, weight_shape).astype("float32")}
+    result_dict = {}
+    for use_trt in [True, False]:
+        x = relay.var("x", shape=x_shape, dtype="float32")
+        weights = relay.var("weights", shape=weight_shape, dtype="float32")
+        out = relay.nn.dense(x, weights)
+        out2 = relay.reshape(out, (-1))
+        f = relay.Function([x, weights], out2)
+        mod = tvm.IRModule()
+        mod["main"] = f
+        if use_trt:
+            mod = relay.tensorrt.EnableTrt(mod, params)
+        with relay.build_config(opt_level=3):
+            relay_exec = relay.create_executor("vm", mod=mod, ctx=tvm.cpu(0), target="llvm")
+
+        for i, batch_size in enumerate(batches_to_test):
+            result_dict[(i, use_trt)] = relay_exec.evaluate()(x=x_data[:batch_size, ...], **params)
+
+    for i in range(len(batches_to_test)):
+        assert_result_matches(result_dict[(i, True)], result_dict[(i, False)])
+
+
 if __name__ == "__main__":
     test_tensorrt_ops()
     test_tensorrt_simple()
     test_tensorrt_simple_cpu_io()
     test_tensorrt_not_compatible()
     test_tensorrt_integration()
     test_tensorrt_serialize()
+    test_tensorrt_dynamic_batch()
```

Comment:
We probably don't need all 4 of these tests. I think just conv2d is good enough.

Reply:
dynamic_batch()mand dynamic_batch_conv() go through different passes, so although we technically dont need dynamic_batch() , do you think its a good idea to just keep it since it goes through a different pass than dynamic_batch_conv()

Reply:
Sure that is fine